core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java

```diff
@@ -466,6 +466,16 @@ public List<Namespace> listNamespaces(Namespace namespace) throws NoSuchNamespac
                         Arrays.stream(n.levels())
                             .limit(subNamespaceLevelLength)
                             .toArray(String[]::new)))
+            // exclude fuzzy matches
+            .filter(
+                n -> {
+                  for (int i = 0; i < namespace.levels().length; i++) {
+                    if (!n.levels()[i].equals(namespace.levels()[i])) {
```

Comment:
this may be more expensive that deduplication, so let's put this filter after the distinct.

Reply:
@Fokko Why did you swap the order in a new commit?

Reply:
This was still open, and I skimmed over it. From intuition, I assume that deduplicating is more expensive than filtering, so I pushed that down to get this in for 1.6.0. I'll revert the change and merge this. Sorry about this